apps/bynder/src/index.js

```diff
@@ -43,51 +66,94 @@ function makeThumbnail(resource) {
 }
 
 function prepareBynderHTML({ bynderURL, assetTypes }) {
-  let types = '';
-  if (!assetTypes) {
-    // We deault to just images in this fallback since this is the behavior the App had in its initial release
-    types = 'image';
-  } else {
-    types = assetTypes.trim().split(',').map(type => type.trim()).join(',');
-  }
-
   return `
     <div class="dialog-container">
-      <div
-        id="bynder-compactview"
-        data-assetTypes="${types}"
-        data-autoload="true"
-        data-button="Load media from bynder.com"
-        data-collections="true"
-        data-folder="bynder-compactview"
-        data-fullScreen="true"
-        data-header="false"
-        data-language="en_US"
-        data-mode="multi"
-        data-zindex="300"
-        data-defaultEnvironment="${bynderURL}"
-      />
-    </div>
+      <div id="bynder-compactview" />
+    </div>      
   `;
 }
 
+function transformAsset(asset) {
+  const thumbnails = {
+    "webimage": asset.files.webImage?.url,
+    "thul": asset.files.thumbnail?.url
+  }
+
+  Object.entries(asset.files)
+      .filter(([name]) => !["webImage", "thumbnail"].includes(name))
+      .forEach(([key, value]) => thumbnails[key] = value?.url);
+
+  return ({
+    "id": asset.databaseId,
+    "orientation": asset.orientation.toLowerCase(),
+    "archive": asset.isArchived ? 1 : 0,
+    "type": asset.type.toLowerCase(),
+    "fileSize": asset.fileSize,
+    "description": asset.description,
+    "name": asset.name,
+    "height": asset.height,
+    "width": asset.width,
+    "copyright": asset.copyright,
+    "extension": asset.extensions,
+    "userCreated": asset.createdBy,
+    "datePublished": asset.publishedAt,
+    "dateCreated": asset.createdAt,
+    "dateModified": asset.updatedAt,
+    "watermarked": asset.isWatermarked ? 1 : 0,
+    "limited": asset.isLimitedUse ? 1 : 0,
+    "isPublic": asset.isPublic ? 1 : 0,
+    "brandId": asset.brandId,
+    "thumbnails": thumbnails
+  })
+}
+
+function checkMessageEvent(e) {
+  if (e.origin === 'https://app.contentful.com') {
```

Comment:
Wont this ignore other environments? i.e: we're assuming it will only work for production environments. Maybe passing in an env variable might be a better idea

Reply:
I flipped the check now, so it should work for all environments :)